cylc/uiserver/tests/test_logging_util.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from glob import glob
+import os
+import pytest
+
+from pathlib import Path
+
+from cylc.uiserver.logging_util import RotatingUISFileHandler
+
+
+def test_update_log_archive(tmp_path):
+    """Test update log archive retains log count at 5"""
+    LOG = RotatingUISFileHandler()
+    LOG.file_path = Path(tmp_path/'.cylc'/'uiserver'/'log')
+    LOG.file_path.mkdir(parents=True, exist_ok=True)
+    for file in [
+        '03-uiserver.log',
+        '01-uiserver.log',
+        '04-uiserver.log',
+        '02-uiserver.log',
+        '05-uiserver.log',
+        '07-uiserver.log',
+        '06-uiserver.log'
+    ]:
```

Comment:
```suggestion
    for file in [f'{i:d02}-uiserver.log' for i in range(1,7)]
```

Reply:
I was making them intentionally out of numerical order.